mesonbuild/compilers/mixins/gnu.py

```diff
@@ -396,3 +396,11 @@ def get_lto_compile_args(self, *, threads: int = 0, mode: str = 'default') -> T.
         elif threads > 0:
             return [f'-flto={threads}']
         return super().get_lto_compile_args(threads=threads)
+
+    @classmethod
+    def use_linker_args(cls, linker: str) -> T.List[str]:
+        if hasattr(cls, 'version') and linker == 'mold':
+            # In detect.py cls is in fact self
+            if mesonlib.version_compare(cls.version, '>=12.1.0'):
+                return ['-fuse-ld=mold']
```

Comment:
`cls` is the idiomatic name for referring to the `Type` object passed in a `classmethod`, while `self` is the idiomatic name of the `object` object passed to an instance method, so this comment isn't super useful.

Also, we shouldn't return something else here, we should error.

Reply:
Of course you are right, I was kind of blind here.
Removed/redesigned.